src/main/java/de/tum/in/www1/artemis/service/ParticipationService.java

```diff
@@ -237,20 +237,68 @@ public StudentParticipation startExercise(Exercise exercise, User user) {
         return participation;
     }
 
+    /**
+     * This method checks whether a participation exists for a given exercise and user. If not, it creates such a participation with initialization state FINISHED.
+     * If the participation had to be newly created or there were no submissions yet for the existing participation, a new submission is created with the given submission type.
+     * For external submissions, the submission is assumed to be submitted immediately upon creation.
+     *
+     * @param exercise the exercise for which to create a participation and submission
+     * @param user the user for which to create a participation and submission
+     * @param submissionType the type of submission to create if none exist yet
+     * @return the participation connecting the given exercise and user
+     */
+    public StudentParticipation createParticipationWithEmptySubmissionIfNotExisting(Exercise exercise, User user, SubmissionType submissionType) {
+        Optional<StudentParticipation> optionalStudentParticipation = findOneByExerciseIdAndStudentLoginAnyState(exercise.getId(), user.getLogin());
+        StudentParticipation participation;
+        if (optionalStudentParticipation.isEmpty()) {
+            // create a new participation only if no participation can be found
+            if (exercise instanceof ProgrammingExercise) {
+                participation = new ProgrammingExerciseStudentParticipation();
+            }
+            else {
+                participation = new StudentParticipation();
+            }
+            participation.setInitializationState(FINISHED);
+            participation.setInitializationDate(ZonedDateTime.now());
+            participation.setExercise(exercise);
+            participation.setStudent(user);
+        }
+        else {
+            // make sure participation and exercise are connected
+            participation = optionalStudentParticipation.get();
+            participation.setExercise(exercise);
+        }
```

Comment:
This is pretty much the same code as for `startExercise`. You could maybe extract some generic participation initialization method here.

Reply:
not sure if we should change `startExercise` here. Even if it overlaps, these are 2 different use cases and they should not overlap too much in the code, otherwise future changes in one of the use case might destroy the other one. I guess we can live with some code duplication

Reply:
While the code is still pretty similar, it has diverged a bit already from the original. I think there is not that much substance in those lines that we should avoid duplication by all means.